### PR TITLE
Fix python validator tests

### LINKED ISSUE #9058: Some python validator tests are broken
Some of the tests defined under https://github.com/cBioPortal/cbioportal/tree/aff9b08c3fa42c128af51a331deedf1cb131e4e6/core/src/test/scripts are broken.

See https://github.com/cBioPortal/cbioportal/runs/4252515774?check_suite_focus=true for the detailed error messages.

Follow up issue for cBioPortal/cbioportal#9054

### PR DESCRIPTION
**Pull request by pvannierop**
Fix #9058

# Problem
Python study validator unit and integration tests do not pass.

# Cause
- The genes.json file was not updated accordingly for the ACPP to ACP3 gene rename.
- result.html files were not updated after data changes to test data.
- Flush of logger in the test file deleted logs resulting in errors in counts.

# Solution
- Genes.json files was updated for ACP3 gene.
- result.html files were updated.
- Test file clones log messages before flushing.

### CODE REVIEW

File: core/src/test/scripts/test_data/study_es_0/result_report.html [hunk 1]
```diff
@@ -134,15 +134,15 @@ <h4 class="panel-title">General</h4>
                 <td><span class="glyphicon glyphicon-cog" aria-hidden="true"></span><span class="sr-only">Debug</span></td>
                 <td>&ndash;</td>
                 <td>&ndash;</td>
-                <td>Retrieving chromosome lengths from &#39;/cbioportal/core/src/main/scripts/importer/chromosome_sizes.json&#39;</td>
+                <td>Retrieving chromosome lengths from &#39;/home/pnp300/git/cbioportal/core/src/main/scripts/importer/chromosome_sizes.json&#39;</td>
```
Comment (SRodenburg) [inline core/src/test/scripts/test_data/study_es_0/result_report.html:137]: this is not correct, the test should be generate from inside the docker container to fix the internal path

File: core/src/test/scripts/test_data/study_es_0/result_report.html [hunk 2]
```diff
@@ -134,15 +134,15 @@ <h4 class="panel-title">General</h4>
...
                 <td>&ndash;</td>
                 <td></td>
               </tr>
               <tr class="info">
                 <td><span class="glyphicon glyphicon-cog" aria-hidden="true"></span><span class="sr-only">Debug</span></td>
                 <td>&ndash;</td>
                 <td>&ndash;</td>
-                <td>Retrieving chromosome lengths from &#39;/cbioportal/core/src/main/scripts/importer/chromosome_sizes.json&#39;</td>
+                <td>Retrieving chromosome lengths from &#39;/home/pnp300/git/cbioportal/core/src/main/scripts/importer/chromosome_sizes.json&#39;</td>
```
Comment (SRodenburg) [inline core/src/test/scripts/test_data/study_es_0/result_report.html:145]: same here

File: core/src/test/scripts/test_data/study_various_issues/result_report.html [hunk 3]
```diff
@@ -299,7 +291,7 @@ <h4 class="modal-title" id="gridSystemModalLabel">Listing all values for the mes
                             <div class="modal-content">
                               <div class="modal-header">
                                 <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
-                                <h4 class="modal-title" id="gridSystemModalLabel">Listing all values for the message: Line will not be loaded due to the variant classification filter. Filtered types: [Silent, Intron, 3'UTR, 3'Flank, 5'UTR, 5'Flank, IGR, RNA]</h4>
+                                <h4 class="modal-title" id="gridSystemModalLabel">Listing all values for the message: Line will not be loaded due to the variant classification filter. Filtered types: [Silent, Intron, 3&#39;UTR, 3&#39;Flank, 5&#39;UTR, 5&#39;Flank, IGR, RNA]</h4>
```
Comment (SRodenburg) [inline core/src/test/scripts/test_data/study_various_issues/result_report.html:294]: is it correct that all non-alpha characters are encoded?

File: core/src/test/scripts/unit_tests_validate_data.py [hunk 4]
```diff
@@ -128,7 +128,8 @@ def validate(self, data_filename, validator_class,
                                     self.logger, relaxed_mode,
                                     strict_maf_checks)
         validator.validate()
-        return self.get_log_records()
```
Comment (SRodenburg) [inline core/src/test/scripts/unit_tests_validate_data.py:131]: I guess this change can be reverted for simplicity

File: core/src/test/scripts/unit_tests_validate_data.py [hunk 5]
```diff
@@ -618,7 +619,7 @@ def test_name_only(self):
         record_list = self.validate('data_cna_genecol_presence_hugo_only.txt',
                                     validateData.CNADiscreteValidator)
         # expecting 1 warning
-        self.assertEqual(len(record_list), 1)
+        self.assertEqual(1, len(record_list))
```
Comment (SRodenburg) [inline core/src/test/scripts/unit_tests_validate_data.py:622]: would be good to fix this throughout

**Review by SRodenburg — CHANGES_REQUESTED** [review]
the paths in the HTML reports should be generalized to the Docker container.

On another note, I am wondering why all HTML text is encoded. Is that correct?

File: core/src/test/scripts/test_data/study_es_0/result_report.html (hunk 1 above)
Comment (pvannierop) [reply core/src/test/scripts/test_data/study_es_0/result_report.html:137]: Fixed

File: core/src/test/scripts/test_data/study_various_issues/result_report.html (hunk 3 above)
Comment (pvannierop) [reply core/src/test/scripts/test_data/study_various_issues/result_report.html:294]: This is done by the validator itself. Tests pass so I guess its no problem.

File: core/src/test/scripts/unit_tests_validate_data.py (hunk 4 above)
Comment (pvannierop) [reply core/src/test/scripts/unit_tests_validate_data.py]: Done.

File: core/src/test/scripts/unit_tests_validate_data.py (hunk 5 above)
Comment (pvannierop) [reply core/src/test/scripts/unit_tests_validate_data.py:621]: Done.

**Review by onursumer — APPROVED** [review]
LGTM. Thanks!

**Comment by pvannierop** [comment]
@onursumer Can we merge this PR?

**Pull request merged**